README.md
# ChatterMap

![Node.js](https://img.shields.io/badge/Node.js-43853D?style=for-the-badge&logo=node.js&logoColor=white) 
![React Native](https://img.shields.io/badge/React_Native-20232A?style=for-the-badge&logo=react&logoColor=61DAFB) 
![ESLint](https://img.shields.io/badge/ESLint-4B3263?style=for-the-badge&logo=eslint&logoColor=white)
![Prettier](https://img.shields.io/badge/Prettier-F7B93E?style=for-the-badge&logo=prettier&logoColor=white)
![GitHub Actions](https://img.shields.io/badge/GitHub_Actions-2088FF?style=for-the-badge&logo=github-actions&logoColor=white)
![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg?style=for-the-badge)


## Getting Started

To get the project up and running on your local machine, follow these steps:

1. **Install dependencies:**
    ```bash
    yarn install
    ```

2. **Start the development server:**
    ```bash
    yarn start
    ```

Now you should have the development server running and can start exploring the project!

## Documentation

- [Change Log](CHANGELOG.md): Track the evolution of the project.
- [Tech Spec](https://chattermap.notion.site/Tech-Spec-7a4f390044de40dcafd913281694b1cf?pvs=4): Dive deeper into the technical landscape of ChatterMap.
- [HISTORY](history.md): See the commit history.

## Backend Library

Our backend services are handled by a separate library. You can find the backend repository [here](https://github.com/ignacio-urrutia/MyApp-backend).

## Contributing

Feel free to submit issues and enhancement requests.

## License

This project is licensed under the MIT License - see the [LICENSE.md](LICENSE.md) file for details.


history.md
commit 344e15ef7b4a4236233ee56803ec2319d4b6d7eb
Merge: 8cf13a1 e0c00fe
Author: Lujie Ma <xiaolinzzz2002@gmail.com>
Date:   Sun Oct 15 18:47:54 2023 +1100

    Merge pull request #33 from XiaoLinZzz/fix/use-token
    
    use token in map

commit e0c00fe99199937f65c9dcd803d02fa0cc5b77eb
Author: Lujie Ma <xiaolinzzz2002@gmail.com>
Date:   Sun Oct 15 18:47:28 2023 +1100

    use token in map

commit 8cf13a1c3a889b1e1803def3d6305a87c45814d8
Merge: dd6c5b0 94b2de3
Author: fcxsdyb <77377800+fcxsdyb@users.noreply.github.com>
Date:   Sun Oct 15 18:34:56 2023 +1100

    Merge pull request #32 from XiaoLinZzz/signup
    
    change signup page name

commit 94b2de312f91bf7b467f58ec28abc1fb04d8895c
Author: Bob <zhouyu@kean.edu>
Date:   Sun Oct 15 18:34:36 2023 +1100

    change signup page name

commit dd6c5b08f61ab33ea71f9c11d972d1c9a221146a
Merge: 538f00d 71e1c1e
Author: Dreamer-T <52307002+Dreamer-T@users.noreply.github.com>
Date:   Sun Oct 15 17:46:55 2023 +1100

    Merge pull request #31 from XiaoLinZzz/setting2.1
    
    Setting2.1

commit 538f00d670f4fc3456ae122e5ce089fe959ffd0b
Merge: a5c6ebb a5457d8
Author: fcxsdyb <77377800+fcxsdyb@users.noreply.github.com>
Date:   Sun Oct 15 17:45:51 2023 +1100

    Merge pull request #30 from XiaoLinZzz/PathChange
    
    Totally successful login and signup page

commit a5457d841d30a0dfffc5d27824287650663eff78
Author: Bob <zhouyu@kean.edu>
Date:   Sun Oct 15 17:43:51 2023 +1100

    Totally successful login and signup page

commit 71e1c1eb13799da20672cd301f3f3333764a157b
Author: Dreamer-T <1727648092@qq.com>
Date:   Sun Oct 15 17:43:29 2023 +1100

    update logout

commit a5c6ebbee7fe078a1e1152df773b44dc3a9a5367
Merge: cd575dc 535a017
Author: Lujie Ma <xiaolinzzz2002@gmail.com>
Date:   Sun Oct 15 17:43:20 2023 +1100

    Merge pull request #29 from XiaoLinZzz/feat/load-other-users-location
    
    load-other-users-location

commit 535a017d2b0ccf3660bc1904f54ea00b6f4c5614
Author: Lujie Ma <xiaolinzzz2002@gmail.com>
Date:   Sun Oct 15 17:40:53 2023 +1100

    load-other-users-location

commit a68a4a7f7bb6b7d101e33efc5b67d263475e4aec
Author: Dreamer-T <1727648092@qq.com>
Date:   Sun Oct 15 17:41:22 2023 +1100

    setting update

commit cd575dc3697b31dd144d31666eddf306220ceadb
Merge: d7a856c a567421
Author: Dreamer-T <52307002+Dreamer-T@users.noreply.github.com>
Date:   Sun Oct 15 17:35:16 2023 +1100

    Merge pull request #28 from XiaoLinZzz/settring2.0
    
    setting update

commit a567421ebfee0a1999ea549df7feb912a2e33167
Author: Dreamer-T <1727648092@qq.com>
Date:   Sun Oct 15 17:34:19 2023 +1100

    setting update

commit d7a856c82e8279b86cb11e196ad85a541e9e767a
Merge: 95d33fc 597a594
Author: Dreamer-T <52307002+Dreamer-T@users.noreply.github.com>
Date:   Sun Oct 15 16:04:19 2023 +1100

    Merge pull request #27 from XiaoLinZzz/private_chatroom_update
    
    Update Chatroom.js

commit 597a594fd5b095fd8f049c2fdda90a4de0e05c5c
Author: Dreamer-T <52307002+Dreamer-T@users.noreply.github.com>
Date:   Sun Oct 15 16:03:54 2023 +1100

    Update Chatroom.js

commit 95d33fc177a4cd8b482b797fbd1eea2ddab45701
Merge: cbd5637 18bd327
Author: fcxsdyb <77377800+fcxsdyb@users.noreply.github.com>
Date:   Sun Oct 15 16:03:05 2023 +1100

    Merge pull request #26 from XiaoLinZzz/PathChange
    
    Document path and correct name

commit 18bd3274d09bb81671b370c97a496709cd8fdb9a
Author: Bob <zhouyu@kean.edu>
Date:   Sun Oct 15 16:02:24 2023 +1100

    Document path and correct name

commit cbd563778ccd2a66693122562f9c67cc6921a3f1
Merge: 6e20555 629dbf5
Author: Lujie Ma <xiaolinzzz2002@gmail.com>
Date:   Sun Oct 15 15:31:06 2023 +1100

    Merge pull request #25 from XiaoLinZzz/fix/comment-clipboard
    
    comment the clipboard

commit 629dbf57be8a43a1a061e12433af90630746505d
Author: Lujie Ma <xiaolinzzz2002@gmail.com>
Date:   Sun Oct 15 15:28:38 2023 +1100

    comment the clipboard

commit 6e205553c27bd08df9b5fd11d6c227c19afe08d0
Merge: 40b5e2c 922a05b
Author: Dreamer-T <52307002+Dreamer-T@users.noreply.github.com>
Date:   Wed Oct 11 19:45:45 2023 +1100

    Merge pull request #24 from XiaoLinZzz/chatroom
    
    chat part init, add vibration setting

commit 922a05b62f0c05af3520a7a21f73fc90676ada9f
Author: Dreamer-T <1727648092@qq.com>
Date:   Wed Oct 11 19:44:07 2023 +1100

    chat part init, add vibration setting

commit 40b5e2cfe2c1681a4faa072af3388f4ca8c937a6
Merge: 8048167 34d6422
Author: Lujie Ma <xiaolinzzz2002@gmail.com>
Date:   Wed Oct 11 14:29:57 2023 +1100

    Merge pull request #23 from XiaoLinZzz/change/reconstruct-the-folder
    
    reconstruct the profile folder

commit 34d64222185ab89c57faad657f966f901ba624b9
Author: Lujie Ma <xiaolinzzz2002@gmail.com>
Date:   Wed Oct 11 14:29:35 2023 +1100

    reconstruct the profile folder

commit 804816798c5ddde7d6a60d70e7bc1e2cfc617cfe
Merge: c4c3cd1 622740e
Author: Lujie Ma <xiaolinzzz2002@gmail.com>
Date:   Wed Oct 11 14:15:53 2023 +1100

    Merge pull request #22 from XiaoLinZzz/newHomepage
    
    Using token to login the page

commit 622740efd5de2adb66354be6a43a88806f3f5ce7
Author: Bob <zhouyu@kean.edu>
Date:   Wed Oct 11 11:29:16 2023 +1100

    Using token to login the page

commit c4c3cd1b55f4be2d9fddb349be952c519ddb6966
Merge: 260319f 3c5ea74
Author: Lujie Ma <xiaolinzzz2002@gmail.com>
Date:   Wed Oct 11 01:27:09 2023 +1100

    Merge pull request #20 from XiaoLinZzz/newHomepage
    
    Change the style of login page and sign up page

commit 3c5ea7406f69f6f835d818392fe610c452adf3f3
Merge: e54785c 260319f
Author: Lujie Ma <xiaolinzzz2002@gmail.com>
Date:   Wed Oct 11 01:27:03 2023 +1100

    Merge branch 'master' into newHomepage

commit 260319fbb2ac72533ff3a8f7196476ac27f60742
Merge: 97810d5 0342e32
Author: Lujie Ma <xiaolinzzz2002@gmail.com>
Date:   Wed Oct 11 01:21:23 2023 +1100

    Merge pull request #21 from XiaoLinZzz/changed/reconstruct-the-navigators-and-some-ui-of-login-signup
    
    reconstruct-the-navigators-and-some-ui-of-login-signup


pages/ChatScreen.js
import React, { useState, useEffect } from 'react';
import { View, Text, Button, FlatList, TextInput, TouchableOpacity, StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { getGroupChatInformation } from '../Services/GroupChatService';

const Stack = createNativeStackNavigator();


const chatRooms = [
  { id: 1, name: "Chat Room 1", description: "sfsdf" },
  { id: 2, name: "Chat Room 2", description: "sfasdfasdfasdfsadfsadsdf" },
  // Add more chat rooms as needed
];

function MainChatScreen() {
  const navigation = useNavigation();
  const [name, setName] = useState("");
  const [chatroomData, setData] = useState()
  useEffect(() => {
    async function test() {
      try {
        const data = await getGroupChatInformation(1);
        setData(data)
        // setName(data.name);
        // console.log(data)
      } catch (error) {
        console.error("Failed to fetch user data:", error);
      }
    }

    test();
  }, []);

  const joinChatRoom = (chatRoomId) => {
    navigation.navigate('ChatRoom', { chatRoomId });
  };
  const renderGroupChatItem = ({ item }) => (
    <TouchableOpacity
      style={styles.groupChatItem}
      onPress={() => joinChatRoom(item)}
    >
      <Text style={styles.groupChatName}>{item.name}</Text>
      <Text style={styles.groupChatMembers}>{item.description}</Text>
    </TouchableOpacity>
  )
  return (
    <View>
      <FlatList
        data={[chatroomData]}
        keyExtractor={(item) => item.id.toString()}
        renderItem={renderGroupChatItem}
      />
    </View>
  );
}

function ChatRoomScreen({ route }) {
  const navigation = useNavigation();
  const { chatRoomId } = route.params;

  const [messages, setMessages] = React.useState([]);
  const [newMessage, setNewMessage] = React.useState('');
  const addMessage = (text, sender) => {
    setMessages((prevMessages) => [
      ...prevMessages,
      { id: prevMessages.length, text, sender },
    ]);
  };

  // change the title of the chat room
  React.useEffect(() => {
    // 名字有问题
    navigation.setOptions({ title: `Chat Room ${chatRoomId}` });
  }, [chatRoomId]);


  const sendMessage = () => {
    if (newMessage) {
      addMessage(newMessage, 'user');
      setNewMessage('');
    }
  };

  return (
    <View style={{ flex: 1, padding: 10 }}>
      <FlatList
        data={messages}
        keyExtractor={(item) => item.id.toString()}
        renderItem={({ item }) => (
          <View
            style={{
              alignSelf: item.sender === 'user' ? 'flex-end' : 'flex-start',
              backgroundColor: item.sender === 'user' ? '#007AFF' : '#E5E5EA',
              borderRadius: 10,
              margin: 5,
              maxWidth: '70%',
              padding: 10,
            }}
          >
            <Text style={{ color: item.sender === 'user' ? 'white' : 'black' }}>
              {item.text}
            </Text>
          </View>
        )}
      />
      <View style={{ flexDirection: 'row', alignItems: 'center' }}>
        <TextInput
          style={{ flex: 1, borderWidth: 1, marginRight: 5, padding: 10, borderRadius: 10 }}
          value={newMessage}
          onChangeText={setNewMessage}
          multiline={true}
        />
        <Button title="send" onPress={sendMessage} />
      </View>
    </View>
  );
}

function ChatScreen() {
  return (
    <Stack.Navigator>
      <Stack.Screen name="Home" component={MainChatScreen} />
      <Stack.Screen name="ChatRoom" component={ChatRoomScreen} />
    </Stack.Navigator>
  );
}

export default ChatScreen;
const styles = StyleSheet.create({
  groupChatItem: {
    paddingHorizontal: 16,
    paddingVertical: 12,
    borderBottomWidth: 1,
    borderBottomColor: '#ddd'
  },
  groupChatName: {
    fontSize: 18,
    fontWeight: 'bold'
  },
  groupChatMembers: {
    fontSize: 14,
    color: '#888'
  },
  headerTitle: {
    fontSize: 24,
    fontWeight: 'bold',
    color: 'white',
    flex: 1
  }
})